solr/core/src/java/org/apache/solr/core/SolrConfig.java

```diff
@@ -158,6 +166,27 @@ public static SolrConfig readFromResourceLoader(SolrResourceLoader loader, Strin
       throw new SolrException(ErrorCode.SERVER_ERROR, "Error loading solr config from " + resource, e);
     }
   }
+  private class ResourceProvider implements Function<String, InputStream> {
+    int zkVersion;
+    int hash = -1;
+    InputStream in;
+    String fileName;
+
+    ResourceProvider(InputStream in) {
+      this.in = in;
+      if (in instanceof ZkSolrResourceLoader.ZkByteArrayInputStream) {
+        ZkSolrResourceLoader.ZkByteArrayInputStream zkin = (ZkSolrResourceLoader.ZkByteArrayInputStream) in;
+        zkVersion = zkin.getStat().getVersion();
+        hash = Objects.hash(zkVersion, overlay.getZnodeVersion());
+        this.fileName = zkin.fileName;
+      }
+    }
+
+    @Override
+    public InputStream apply(String s) {
+      return in;
+    }
```

Comment:
Why do we implement Function if we don't use the argument? And always return the same input stream? Where does this get called from?